### PR TITLE
ensure hard disk is encrypted

### PR DESCRIPTION
**Pull request by johnduffell**
Everyone's hard disk should be encrypted, this just adds a check to make sure it's so before downloading the properties file from S3.

Not sure how to do it on linux - someone who knows might update that side of the script?

@TBonnin 
@adamnfish @janua you are linux guys, what do you think?

### CODE REVIEW

**Comment by dominickendrick** [comment]
Great stuff !

File: setup.sh [hunk 1]
```diff
@@ -56,8 +56,13 @@ create_frontend_properties() {
     if [[ ! -d "$path" ]]; then
       mkdir "$path"
     fi
-
-    aws s3 cp --profile frontend s3://aws-frontend-store/template-frontend.properties "$path/$filename"
+    
+    if [[ "$(fdesetup status)" != "FileVault is On." ]]; then
+      echo your hard disk is not encrypted!
```
Comment (TBonnin) [inline setup.sh]: I will also tell people what to do to solve the problem ex: `If Mac, go to Preferences > Security > FileVault ... If Linux...`

Comment (johnduffell) [inline setup.sh]: no I'm hoping the linux people will  let me know what command to add for that =)

**Comment by adamnfish** [comment]
:-1: needs a `uname contains darwin` check. This will break it for everyone not on OSX!

**Comment by johnduffell** [comment]
@adamnfish 
`if uname != darwin; then echo "please write the correct script to check encryption is on before continuing"; exit 1; fi`

**Comment by janua** [comment]
Yeah definitely needs a check for linux at least, warning that it isn't checked.

Maybe there is a more OS free way of checking it? I know when you list mounts they say if they are encrypted or not in linux, not sure about mac though.

**Comment by desbo** [comment]
the script already has OS checks, you can just reuse them with `if linux` or `if mac`

File: setup.sh [hunk 2]
```diff
@@ -56,8 +56,13 @@ create_frontend_properties() {
     if [[ ! -d "$path" ]]; then
       mkdir "$path"
     fi
-
-    aws s3 cp --profile frontend s3://aws-frontend-store/template-frontend.properties "$path/$filename"
+    
+    if [[ "$(fdesetup status)" != "FileVault is On." ]]; then
+      echo your hard disk is not encrypted! follow these instructions: https://support.apple.com/en-gb/HT204837
```
Comment (desbo) [inline setup.sh]: you can use the `EXTRA_STEPS` thing here, have a look at the other examples in the file

**Comment by sndrs** [comment]
:floppy_disk:

**Comment by johnduffell** [comment]
ok thanks for the comments, I guess there's no simple linux check so I'll just skip that bit, someone can add it later!  Take another look 😛

**Comment by desbo** [comment]
lgtm 👍

**Pull request merged**